src/suricata.c

```diff
@@ -407,6 +410,8 @@ static void GlobalsDestroy(SCInstance *suri)
     SCThresholdConfGlobalFree();
 
     SCPidfileRemove(suri->pid_filename);
+    free((char *)suri->pid_filename);
```

Comment:
Note that you can only call `free` if the memory you're returning has been allocated by `malloc` (or equivalent).